modules/integration/src/test/scala/scala/cli/integration/RunScalacCompatTestDefinitions.scala

```diff
@@ -206,6 +206,43 @@ trait RunScalacCompatTestDefinitions {
     }
   }
 
+  test("scala-cli compile -d option package into a jar") {
+    val preCompileDir = "PreCompileDir"
+    val mainInput = "Main.scala"
+    val expectedOutput = "Hello"
+    TestInputs(
+      os.rel / preCompileDir / mainInput ->
+        s"""object Main extends App { println("$expectedOutput") }""".stripMargin
+    ).fromRoot { (root: os.Path) =>
+      val outputJar = os.rel / "outParentDir" / "out.jar"
+
+      // first, precompile to an explicitly specified output jar file with -d
+      os.proc(
+        TestUtil.cli,
+        "compile",
+        "-p",
+        mainInput,
+        "-d",
+        outputJar.toString,
+        extraOptions
+      ).call(cwd = root / preCompileDir)
+
+      // ensure the precompiled JAR exists and is a file ( not a directory )
+      assert(os.exists(root / preCompileDir / outputJar), s"Precompiled JAR not found at $outputJar")
+      assert(!os.isFile(root / preCompileDir / outputJar), s"$outputJar is not a file")
```

Comment:
```suggestion
      assert(os.isFile(root / preCompileDir / outputJar), s"$outputJar is not a file")
```
this matches up with the comments and assertion message

Reply:
this assertion is passing, because the produced outputJar is a directory, rather than a JAR file.
the directory contains compiled classes, so passing them as an inputs does, in fact, work.
it does not fulfill the requirements of what we want here, however.
```
run-891283530/test-2/PreCompileDir
▶ cd outParentDir 

test-2/PreCompileDir/outParentDir 
▶ ls
out.jar

test-2/PreCompileDir/outParentDir
▶ cd out.jar

PreCompileDir/outParentDir/out.jar
▶ ls
Main$.class Main.class  Main.tasty
```